components/brave_new_tab_ui/components/default/rewards/index.tsx

```diff
@@ -0,0 +1,274 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this file,
+* You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import createWidget from '../widget/index'
+import { convertBalance, tipsTotal } from '../../../../brave_rewards/resources/page/utils'
+import { getLocale } from '../../../../common/locale'
+
+import {
+  WidgetWrapper,
+  BatIcon,
+  RewardsTitle,
+  Footer,
+  ServiceText,
+  ServiceLink,
+  LearnMoreText,
+  PreOptInInfo,
+  Title,
+  SubTitle,
+  PreOptInAction,
+  TurnOnButton,
+  AmountItem,
+  AmountInformation,
+  AmountDescription,
+  Amount,
+  ConvertedAmount,
+  LearnMoreLink,
+  TurnOnAdsButton
+} from './style'
+import Notification from './notification'
+import { BatColorIcon } from 'brave-ui/components/icons'
+
+export interface RewardsProps {
+  enabledAds: boolean
+  enabledMain: boolean
+  balance: NewTab.RewardsBalance
+  grants: NewTab.GrantRecord[]
+  reports: Record<string, NewTab.RewardsReport>
+  walletCreated: boolean
+  walletCreating: boolean
+  walletCreateFailed: boolean
+  walletCorrupted: boolean
+  adsEstimatedEarnings: number
+  onCreateWallet: () => void
+  onEnableAds: () => void
+  onEnableRewards: () => void
+  onDismissNotification: (id: string) => void
+}
+
+class Rewards extends React.PureComponent<RewardsProps, {}> {
+
+  getTotalContributions = () => {
+    const currentTime = new Date()
+    const reportKey = `${currentTime.getFullYear()}_${currentTime.getMonth() + 1}`
+    const report: NewTab.RewardsReport = this.props.reports[reportKey]
+
+    if (report) {
+      return tipsTotal(report)
+    }
+
+    return '0.0'
+  }
+
+  getButtonText = (isAds: boolean = false) => {
+    if (isAds) {
+      return getLocale('turnOnAds')
```

Comment:
why some locals has prefix `rewardsWidget` and some not?

Reply:
i've made them consistent, i originally did this because some of these locales are used/will be used in domains besides the widget.